app/controllers/notes_controller.rb

```diff
@@ -1,7 +1,11 @@
 class NotesController < ApplicationController
   def index
     @user = current_user
-    @notes = @user.notes
+    @notes = if params[:search_key].present?
```

Comment:
[minor] You could extract `params[:search_key]` into a method called `search_key` for readability since it's being looked up multiple times.

Reply:
I will define a private function called `search_key` in notes_controller.